clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -696,7 +696,12 @@ public void onLeavePrepare() {
         Set<TopicPartition> droppedPartitions = new HashSet<>(subscriptions.assignedPartitions());
 
         if (subscriptions.partitionsAutoAssigned() && !droppedPartitions.isEmpty()) {
-            final Exception e = invokePartitionsRevoked(droppedPartitions);
+            final Exception e;
+            if (generation() != Generation.NO_GENERATION) {
```

Comment:
@abbccdda Just wondering, have you tried running the failing `test_broker_type_bounce` without this optimization?

Reply:
No I haven't, that's a good idea.